libraries/AP_Math/scurves.cpp

```diff
@@ -0,0 +1,645 @@
+        return move_from_pva_straight(dt, time_scale, pos, vel, accel);
+    } else if (segtype == SegmentType::SPLINE) {
+        return move_from_pva_spline(dt, time_scale, pos, vel, accel);
+    } else {
+        return true;
+    }
+}
+
+// return the position, velocity and acceleration vectors relative to the destination
+bool scurves::move_to_pos_vel_accel(float dt, float time_scale, Vector3f &pos, Vector3f &vel, Vector3f &accel)
+{
+    if (segtype == SegmentType::STRAIGHT) {
+        return move_to_pva_straight(dt, time_scale, pos, vel, accel);
+    } else if (segtype == SegmentType::SPLINE) {
+        return move_to_pva_spline(dt, time_scale, pos, vel, accel);
+    } else {
+        return true;
+    }
+}
+
+// return the position, velocity and acceleration vectors relative to the origin
+bool scurves::move_from_time_pos_vel_accel(float time, float time_scale, Vector3f &pos, Vector3f &vel, Vector3f &accel)
+{
+    if (segtype == SegmentType::STRAIGHT) {
+        return move_from_time_pva_straight(time, time_scale, pos, vel, accel);
+    } else if (segtype == SegmentType::SPLINE) {
+        return move_from_time_pva_spline(time, time_scale, pos, vel, accel);
+    } else {
+        return true;
+    }
+}
+
+// return true if the current segment is a straight segment
+bool scurves::is_straight() const
+{
+    return segtype == SegmentType::STRAIGHT;
+}
+
+// magnitude of the position vector at the end of the sequence
+float scurves::pos_end() const
+{
+    if (segtype == SegmentType::STRAIGHT) {
+        return pos_end_straight();
+    } else if (segtype == SegmentType::SPLINE) {
+        return pos_end_spline();
+    } else {
+        return 0.0f;
+    }
+}
+
+// time at the end of the sequence
+float scurves::time_end() const
+{
+    if (segtype == SegmentType::STRAIGHT) {
+        return time_end_straight();
+    } else if (segtype == SegmentType::SPLINE) {
+        return time_end_spline();
+    } else {
+        return 0.0f;
+    }
+}
+
+// time left before sequence will complete
+float scurves::get_time_remaining() const
+{
+    if (segtype == SegmentType::STRAIGHT) {
+        return get_time_remaining_straight();
+    } else if (segtype == SegmentType::SPLINE) {
+        return get_time_remaining_spline();
+    } else {
+        return 0.0f;
+    }
+}
+
+// return true if the sequence is braking to a stop
+bool scurves::braking() const
+{
+    if (segtype == SegmentType::STRAIGHT) {
+        return braking_straight();
+    } else if (segtype == SegmentType::SPLINE) {
+        return braking_spline();
+    } else {
+        return true;
+    }
+}
+
+// Straight implementations
+
+// increment time pointer and return the position, velocity and acceleration vectors relative to the origin
+bool scurves::move_from_pva_straight(float dt, float time_scale, Vector3f &pos, Vector3f &vel, Vector3f &accel)
+{
+    advance_time(time_scale * dt);
+    bool finish = move_from_time_pva_straight(_t, time_scale, pos, vel, accel);
```

Comment:
```suggestion
    const bool finish = move_from_time_pva_straight(_t, time_scale, pos, vel, accel);
```

Reply:
code has changed but in any case, I think we have added "const" everywhere we can.